### PR TITLE
Columns should have a sortValue prop, Fixes: #2396

### PR DESCRIPTION
**Pull request by febbraiod**
Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

**Comment by febbraiod** [comment]
Can test with : https://github.com/xh/toolbox/tree/SortValue

File: cmp/grid/columns/Column.js [hunk 1]
```diff
@@ -548,8 +562,20 @@ export class Column {
...
+            recordA = agNodeA?.data,
+            recordB = agNodeB?.data;
+
+        if (isString(sortValue)) {
+            v1 = recordA?.data[sortValue] ?? v1;
+            v2 = recordB?.data[sortValue] ?? v2;
+        } else if (isFunction(sortValue)) {
+            v1 = sortValue(v1, {recordA, column: this, gridModel});
+            v2 = sortValue(v2, {recordB, column: this, gridModel});
+        }
```
Comment (TomTirapani) [inline cmp/grid/columns/Column.js:577]: This code block is very similar to the one above. Could we DRY these up with a `getSortValue(record)` method?

File: cmp/grid/columns/Column.js [hunk 2]
```diff
@@ -532,6 +536,16 @@ export class Column {
                         agNodeB
                     };
 
+                if (isString(sortValue)) {
+                    // If sortValue points to a different field
+                    valueA = recordA?.data[sortValue] ?? valueA; // Group rows have no record.
```
Comment (lbwexler) [inline cmp/grid/columns/Column.js:541]: this note about groupRows seems problematic?  So will this even work or a groupedGrid that is trying to use the new sortValue property?

Comment (febbraiod) [reply cmp/grid/columns/Column.js:541]: Well the value for the grouped row was being passed to the comparator before regardless. It was an empty string. Which it remains here.

File: cmp/grid/columns/Column.js (hunk 1 above)
Comment (febbraiod) [reply cmp/grid/columns/Column.js:577]: yes good idea.

File: cmp/grid/columns/Column.js (hunk 2 above)
Comment (febbraiod) [reply cmp/grid/columns/Column.js:541]: Tested whilst paying attn to the group rows. The sort works, but the group rows now jump around with the when sorting by this column. Previously the groups would stay put. Will revisit.

**Comment by lbwexler** [comment]
great -- eager to merge this when you work this out.  Was it sorting the
groups based on 'null', or some other sort value of the group, in that case?

**Comment by febbraiod** [comment]
Was using an empty string for the group rows. And I guess since "" === "" the group rows stay put. Just want to look at together for a sanity check since it's gone this far with a bit of confusion.

**Pull request merged**